File: programming problems/HackerRank/30 days of code/day 6, lets review/day_6_lets_review.cpp
#include <iostream>
using namespace std;

int main() {

    int n;
    cin >> n;
    
    for (string word, word1, word2; cin >> word; word1 = word2 = "") {
        
        // c++20 for (auto flag {true}; const auto& ch : word)
        auto flag {true};
        for (const auto& ch : word) {
            flag ? word1.append(1, ch) : word2.append(1, ch);
            flag = !flag;
        }
        
        cout << word1 << ' ' << word2 << '\n';
    }
    return 0;
}

File: programming problems/HackerRank/30 days of code/day 6, lets review/day_6_lets_review.rb
n = gets.chomp.to_i
for i in 1..n do

    word, word1, word2 = gets.chomp, "", ""
    flag = true
    
    word.each_char.with_index do |ch, i|
        flag ? (word1 << ch) : (word2 << ch)
        flag = !flag
    end

    puts "#{word1} #{word2}"
end